tests/README.md

```diff
@@ -0,0 +1,515 @@
+   collected 629 items / 626 deselected / 3 selected
+
+   eth2spec/test/merge/sanity/test_blocks.py .                              [ 33%]
+   eth2spec/test/phase0/sanity/test_blocks.py ..                            [100%]
+
+   =============================== warnings summary ===============================
+   ../../../venv/lib/python3.9/site-packages/cytoolz/compatibility.py:2
+     /home/qbzzt1/consensus-specs/venv/lib/python3.9/site-packages/cytoolz/compatibility.py:2: 
+   DeprecationWarning: The toolz.compatibility module is no longer needed in Python 3 and has 
+   been deprecated. Please import these utilities directly from the standard library. This 
+   module will be removed in a future release.
+       warnings.warn("The toolz.compatibility module is no longer "
+
+   -- Docs: https://docs.pytest.org/en/stable/warnings.html
+   ================ 3 passed, 626 deselected, 1 warning in 16.81s =================   
+   ```
+
+
+## The "Hello, World" of Consensus Spec Tests
+
+One of the `test_empty_block_transition` tests is implemented by a function with the same
+name located in 
+[`~/consensus-specs/tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py`](https://github.com/ethereum/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py).
+To learn how consensus spec tests are written, let's go over the code:
+
+```python
+@with_all_phases
+```
+
+This [decorator](https://book.pythontips.com/en/latest/decorators.html) specifies that this test
+is applicable to all the phases of the ETH 2.0 project. These phases are similar to forks (Istanbul,
+Berlin, London, etc.) in the execution blockchain. If you are interested, [you can see the definition of
+this decorator here](https://github.com/ethereum/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/context.py#L331-L335).
+
+```python
+@spec_state_test
+```
+
+[This decorator](https://github.com/qbzzt/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/context.py#L232-L234) specifies
+that this test is a state transition test, and that it does not include a transition between different forks.
+
+```python
+def test_empty_block_transition(spec, state):
+```
+
+This type of test receives two parameters:
+
+* `specs`: The protocol specifications
+* `state`: The genesis state before the test
+
+```python
+    pre_slot = state.slot
+```    
+
+A slot is a unit of time (every 12 seconds in mainnet), for which a specific validator (selected randomly but in a
+deterministic manner) is a proposer. The proposer can propose a block during that slot. 
+
+```python
+    pre_eth1_votes = len(state.eth1_data_votes)
+    pre_mix = spec.get_randao_mix(state, spec.get_current_epoch(state))
+```
+
+Store some values to check later that certain updates happened.
+
+```python
+    yield 'pre', state
+```
+
+In Python `yield` is used by [generators](https://wiki.python.org/moin/Generators). However, for our purposes
```

Comment:
It would be nice to clarify why we are yielding this. Reason why: the tests can be executed in pytest, but also used as functions in the test-generator script, to create test-vectors from these yielded outputs.

About test generators: https://github.com/ethereum/consensus-specs/blob/dev/tests/generators/README.md
About test formats (what is yielded per type of test): https://github.com/ethereum/consensus-specs/tree/dev/tests/formats (you linked it later in the doc, but it may help clarify these `yield` things)

Reply:
I need to think about it. On one hand extra knowledge is always good. On the other, in a tutorial I try to minimize the cognitive load by having people focus on what they need to know to do the job (plus to answer any questions that appear to me to be something students will think about, so they won't have to figure the answer themselves).

Maybe I should explain in this the section about how these tests are used. But I should think about it when I'm more awake.

Reply:
I linked to the generators README.md. I think that would be enough - whoever wants to read can read.